NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -83,30 +128,39 @@ -(id) init {
 }
 
 - (void) run {
+    self.test_id = [NSNumber numberWithDouble:[[NSDate date] timeIntervalSince1970]];
+    self.json_file = [NSString stringWithFormat:@"test-%@.json", self.test_id];
+    self.log_file = [NSString stringWithFormat:@"test-%@.log", self.test_id];
+    [TestStorage add_test:self];
     setup_idempotent();
     NSBundle *bundle = [NSBundle mainBundle];
     NSString *path = [bundle pathForResource:@"hosts" ofType:@"txt"];
-    mk::ooni::DnsInjectionTest()
-        .set_backend("8.8.8.8:53")
-        .set_input_file_path([path UTF8String])
-        .set_verbose()
-        .on_log([self](const char *s) {
+    mk::ooni::DnsInjection()
+        .set_options("backend", "8.8.8.1:53")
+        .set_options("dns/nameserver", get_dns_server())
+        .set_options("net/ca_bundle_path", [ca_cert UTF8String])
+        .set_options("geoip_country_path", [geoip_country UTF8String])
+        .set_options("geoip_asn_path", [geoip_asn UTF8String])
+        .set_input_filepath([path UTF8String])
+        .set_output_filepath([[self getFileName:@"json"] UTF8String])
+        .set_verbosity(MK_LOG_DEBUG2)
```

Comment:
Note to self: we should probably lower the log level before merging the pull request because this level of verbosity is frankly very high (I think `MK_LOG_INFO` would be much better).